src/node.rs

```diff
@@ -1133,18 +1152,87 @@ impl<S: ForkSource + std::fmt::Debug> InMemoryNode<S> {
+                    l1_batch_number: block.l1_batch_number,
+                    transaction_hash: Some(tx_hash),
+                    transaction_index: Some(U64::zero()),
+                    log_index: Some(U256::from(log_idx)),
+                    transaction_log_index: Some(U256::from(log_idx)),
+                    log_type: None,
+                    removed: None,
+                },
+                U64::from(current_miniblock),
+            );
+        }
+        let tx_receipt = TransactionReceipt {
+            transaction_hash: tx_hash,
+            transaction_index: U64::from(0),
+            block_hash: Some(block.hash),
+            block_number: Some(block.number),
+            l1_batch_tx_index: None,
+            l1_batch_number: block.l1_batch_number,
+            from: l2_tx.initiator_account(),
+            to: Some(l2_tx.recipient_account()),
+            cumulative_gas_used: Default::default(),
+            gas_used: Some(l2_tx.common_data.fee.gas_limit - result.gas_refunded),
+            contract_address: contract_address_from_tx_result(&result),
+            logs: result
+                .result
+                .logs
+                .events
+                .iter()
+                .enumerate()
+                .map(|(log_idx, log)| Log {
+                    address: log.address,
+                    topics: log.indexed_topics.clone(),
+                    data: Bytes(log.value.clone()),
+                    block_hash: Some(block.hash),
+                    block_number: Some(block.number),
+                    l1_batch_number: block.l1_batch_number,
+                    transaction_hash: Some(tx_hash),
+                    transaction_index: Some(U64::zero()),
+                    log_index: Some(U256::from(log_idx)),
+                    transaction_log_index: Some(U256::from(log_idx)),
+                    log_type: None,
+                    removed: None,
+                })
+                .collect(),
+            l2_to_l1_logs: vec![],
+            status: Some(if result.status == TxExecutionStatus::Success {
+                U64::from(1)
+            } else {
+                U64::from(0)
+            }),
+            effective_gas_price: Some(L2_GAS_PRICE.into()),
+            ..Default::default()
+        };
         inner.tx_results.insert(
             tx_hash,
-            TxExecutionInfo {
-                tx: l2_tx,
-                batch_number: block.l1_batch_number.unwrap_or_default().as_u32(),
-                miniblock_number: current_miniblock,
-                result,
+            TransactionResult {
+                info: TxExecutionInfo {
+                    tx: l2_tx,
+                    batch_number: block.l1_batch_number.unwrap_or_default().as_u32(),
+                    miniblock_number: current_miniblock,
+                    result,
+                },
+                receipt: tx_receipt,
             },
         );
+        let block_hash = block.hash;
```

Comment:
Is this variable necessary?

Reply:
The variable `block` is inserted into the `blocks` storage so the hash isn't available when we need to notify it on line `1234`. The alternate would be to clone the entire variable.